src/creator.ts

```diff
@@ -238,6 +238,23 @@ export class SlashCreator extends (EventEmitter as any as new () => TypedEventEm
     this.emit('debug', 'Server started');
   }
 
+  /**
+   * Registers a global component callback. Note that this will have no expiration, and should be invoked by the returned name.
+   * @param custom_id The custom ID of the component to register
+   * @param callback The callback to use on interaction
+   */
+  registerGlobalComponent(custom_id: string, callback: ComponentRegisterCallback) {
+    if (this._componentCallbacks.has(custom_id))
+      throw new Error(`A global component with the ID "${custom_id}" is already registered.`);
+    const newName = `global-${custom_id}`;
```

Comment:
```suggestion
  registerGlobalComponent(custom_id: string, callback: ComponentRegisterCallback) {
    const newName = `global-${custom_id}`;
    if (this._componentCallbacks.has(newName))
      throw new Error(`A global component with the ID "${newName}" is already registered.`);
```

Reply:
I'm sorry, shouldn't have submitted that the way it is. Guess I should drink more coffee before opening a PR :P